datanode/entities/fund_period.go

```diff
@@ -0,0 +1,202 @@
+package entities
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	v2 "code.vegaprotocol.io/vega/protos/data-node/api/v2"
+
+	"code.vegaprotocol.io/vega/libs/ptr"
+
+	events "code.vegaprotocol.io/vega/protos/vega/events/v1"
+
+	"code.vegaprotocol.io/vega/libs/num"
+)
+
+type FundingPeriod struct {
```

Comment:
This should also include:
1. Market price TWAP (time-weighted average price coming from the perps market on Vega
2. Reference price TWAP (time-weighed average price coming from the oracle)

Reply:
This isn't being provided by core, only the data points. The question is should the TWAP market/reference price be provided by core, or do you want this to be calculated from the data points by data node? The entity here represents the database object that we map from the core event we receive so this may or may not be updated depending on the answer.

Reply:
Confirmed with @witgaw that the TWAP information should be emitted from core.